### ISSUE TITLE
Failed to run 'apt update' on ubuntu-base image when /etc/resolv.conf is NFS rooted

### ISSUE THREAD

**Issue by ZQyou** [apptainer/singularity#4666]
### Version of Singularity
```
$ singularity version
3.4.1-1.el7
```

### Expected behavior
```
$ singularity build -f test.sif test.def
INFO:    Starting build...
INFO:    Downloading library image
INFO:    Running post scriptlet
+ apt update
Hit:1 http://us.archive.ubuntu.com/ubuntu bionic InRelease
Reading package lists... Done
[ ... skipped ... ]
INFO:    Build complete: test.sif
```
```
$ cat test.def
BootStrap: library
From: ubuntu:18.04

%post
    apt update
    apt upgrade -y
```
### Actual behavior
```
$ singularity build -f test.sif test.def
NFO:    Starting build...
INFO:    Running post scriptlet
+ apt update
Err:1 http://us.archive.ubuntu.com/ubuntu bionic InRelease
  Could not resolve 'us.archive.ubuntu.com'
[ ... skipped ... ]
INFO:    Build complete: test.sif
```

### Steps to reproduce this behavior
Run `df -h /etc/resolv.conf` in the Singularity file, I get 
```
+ df -h /etc/resolv.conf
Filesystem                                                     Size  Used Avail Use% Mounted on
root.host:/rhel76_0/etc/resolv.conf  190G   86G  105G  45% /etc/resolv.conf
```
If I remount `/etc/resolv.conf` from  local file system, the problem is gone:
```
    cat /etc/resolv.conf > ./resolv.conf
    mount -o bind ./resolv.conf /etc/resolv.conf
```

### What OS/distro are you running
```
$ cat /etc/os-release
NAME="Red Hat Enterprise Linux Server"
VERSION="7.6 (Maipo)"
ID="rhel"
ID_LIKE="fedora"
VARIANT="Server"
VARIANT_ID="server"
VERSION_ID="7.6"
PRETTY_NAME="Red Hat Enterprise Linux"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:7.6:GA:server"
HOME_URL="https://www.redhat.com/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 7"
REDHAT_BUGZILLA_PRODUCT_VERSION=7.6
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="7.6"
```
### How did you install Singularity
RPM

**Comment by dtrudg** [comment]
I believe this should be solved in https://github.com/sylabs/singularity/pull/4629 - where `resolv.conf` is now copied to a `tmpfs` staging area. Are you able to test from `master`? I'm afraid I don't have a particularly quick way to test an nfs rooted setup.

**Comment by ZQyou** [comment]
Thank you. I do see `/etc/resolv.conf`  staged  in `tmpfs` when using `shell -f -w`:
```
$ singularity build --sandbox ubuntu library://library/default/ubuntu
$ singularity shell -f -w ubuntu
Singularity> df -h /etc/resolv.conf
Filesystem      Size  Used Avail Use% Mounted on
tmpfs            16M  4.0K   16M   1% /etc/resolv.conf
```
So I can run `apt update` when using `shell` command. 

Regarding #4629, I can run update on centos images  using  both `shell` and `build` commands. I wonder if something is just missing for ubuntu. 

Our `root host` has no production environment. I cannot run test on this host.

**Comment by dtrudg** [comment]
Hi @ZQyou - I'm slightly confused which base OS you are using, and which Singularity versions in that last message.

Can you show the OS, `singularity version` and your output for those?

Were you using the #4629 PR in all cases, or is that Ubuntu shell output from another version?

**Comment by ZQyou** [comment]
Sorry for confusing. In the previous message, I run `shell` command on the system as I reported at the beginning. 

I did not use  #4629 PR. Just want to say that I can run `yum update` on centos images with the same Singularity version.

**Comment by dtrudg** [comment]
The behavior here is a little bit complex - it depends on the combination of the host OS, the guest OS, and if the host's `resolv.conf` has been modified - since that will all affect what `yum` or `apt` see, and whether they decide to try and update the file.

This should all have been fixed in #4629 - and won't be an issue in the upcoming `3.5` release. If you would like to test, it would need to be by building Singularity from the `master` branch.

**Comment by ZQyou** [comment]
Thanks. We will wait for `3.5` release.

**Comment by cclerget** [comment]
Fixed by #4629
